File: RobWork/example/java/src/ExWorldTransforms.java
import java.util.Iterator;

import org.robwork.sdurw.*;

public class ExWorldTransforms {
    public static Transform3DdVector worldTransforms(
            FrameVector frames, State state)
    {
        FKTable fk = new FKTable(state);

        Transform3DdVector result = new Transform3DdVector();
        Iterator<Frame> iterator = frames.iterator();
        while(iterator.hasNext()) {
            Frame frame = iterator.next();
            result.add(fk.get(frame));
        }
        return result;
    }
}

File: RobWork/example/cpp/ex_world-transforms.cpp
#include <rw/kinematics/FKTable.hpp>

#include <vector>

using rw::math::Transform3D;
using namespace rw::kinematics;

std::vector<Transform3D<> > worldTransforms(
    const std::vector<const Frame*>& frames, const State& state)
{
    FKTable fk(state);

    std::vector<Transform3D<> > result;
    for(const Frame* const f : frames) {
        result.push_back(fk.get(*f));
    }
    return result;
}
